neps/nep-0330.md

```diff
@@ -31,23 +35,32 @@ There is a lot of information that can be held about a contract. Ultimately, we
 - `standards`: a list of objects (see type definition below) that enumerates the NEPs supported by the contract. If this extension is supported, it is advised to also include NEP-330 version 1.1.0 in the list (`{standard: "nep330", version: "1.1.0"}`).
+- `build_info`: a build details object (see type definition below) that contains all the necessary information about how the contract was built, making it possible for others to reproduce the same WASM of this contract.
 
 ```ts
 type ContractSourceMetadata = {
-  version: string|null, // optional, commit hash being used for the currently deployed wasm. If the contract is not open-sourced, this could also be a numbering system for internal organization / tracking such as "1.0.0" and "2.1.0".
-  link: string|null, // optional, link to open source code such as a Github repository or a CID to somewhere on IPFS.
-  standards: Standard[]|null, // optional, standards and extensions implemented in the currently deployed wasm e.g. [{standard: "nep330", version: "1.1.0"},{standard: "nep141", version: "1.0.0"}].
+    version: string|null, // optional, commit hash being used for the currently deployed wasm. If the contract is not open-sourced, this could also be a numbering system for internal organization / tracking such as "1.0.0" and "2.1.0".
+    link: string|null, // optional, link to open source code such as a Github repository or a CID to somewhere on IPFS. e.g. https://"github.com/near-examples/nft-tutorial/tree/39f2d2646f2f60e18ab53337501370dc02a5661c"
+    standards: Standard[]|null, // optional, standards and extensions implemented in the currently deployed wasm e.g. [{standard: "nep330", version: "1.1.0"},{standard: "nep141", version: "1.0.0"}].
+    build_info: BuildInfo|null, // optional, details that are required for contract WASM reproducibility.
 }
 
 type Standard {
     standard: string, // standard name e.g. "nep141"
     version: string, // semantic version number of the Standard e.g. "1.0.0"
 }
+
+type BuildInfo {
+    build_environment: string, // reference to a reproducible build environment docker image, e.g. "sourcescan/cargo-near@sha256:bf488476d9c4e49e36862bbdef2c595f88d34a295fd551cc65dc291553849471" or something else pointing to the build environment. When using a Docker image as a reference, it's important to specify the digest of the image to ensure reproducibility, since a tag could be reassigned to a different image.
+    source_code_snapshot: string, // reference to the source code snapshot that was used to build the contract, e.g. "git+https://github.com/near-DevHub/neardevhub-contract.git#335e89edec95d56a4744e7160c3fd590e41ec38e" or "ipfs://<ipfs-hash>"
+    contract_path: string|null, // relative path to contract crate within the source code e.g. "src/contract". Often, it is the root of the repository, so can be omitted. 
```

Comment:
`src/` might be misleading and readers may think they should somehow reference a file within `src/` folder (standard in Rust projects), so I suggest to use more neutral example:

```suggestion
    contract_path: string|null, // relative path to contract crate within the source code e.g. "contracts/contract-one". Often, it is the root of the repository, so can be omitted. 
```